src/server/auth/server/api_server.go

```diff
@@ -362,6 +362,32 @@ func (a *apiServer) Activate(ctx context.Context, req *auth.ActivateRequest) (re
 	return &auth.ActivateResponse{PachToken: pachToken}, nil
 }
 
+// RotateRootToken implements the protobuf auth.RotateRootToken RPC
+func (a *apiServer) RotateRootToken(ctx context.Context, req *auth.RotateRootTokenRequest) (resp *auth.RotateRootTokenResponse, retErr error) {
+	// We don't want to actually log the request/response since they contain credentials.
+	defer func(start time.Time) { a.LogResp(nil, nil, retErr, time.Since(start)) }(time.Now())
+
+	// TODO(acohen4): Transactionify so that token deletion and insertion are atomic
```

Comment:
Is it a big lift to make this atomic? I would feel better if this was wrapped in a transaction before it hits master, but I might be missing context.

Reply:
nah not a big lift. My take has been that it will be easy to add once postgres-integration is merged. 
I'll add the transaction wrapping and leave a note to migrate it to some of the library code written in postgres-integration